website/backend/src/chainTracker.ts

```diff
@@ -25,117 +25,195 @@ interface TimestampedError {
 	error: unknown
 }
 
-export function startIndexingChainToDataStores(
-	coordinator: ServiceConfig,
-	miningPool: ServiceConfig
-): {
-	cancel: () => void
-	coordinator: {
-		stopped: Promise<void>
-		dataStore: CoordinatorDataStore
-		errors: TimestampedError[]
-	}
-	miningPool: {
-		stopped: Promise<void>
-		dataStore: MiningPoolDataStore
-		errors: TimestampedError[]
-	}
-} {
-	const stateDirectory = process.env.STATE_DIRECTORY ?? process.cwd()
-
-	// create working dir so we can write files to it later
-	mkdirSync(stateDirectory, { recursive: true })
+interface ServiceResult<T> {
+	stopped: Promise<void>
+	dataStore: T
+	errors: TimestampedError[]
+}
 
-	const cancelled = { cancelled: false }
+function discoverExistingCoordinators(stateDirectory: string): string[] {
+	try {
+		const files = readdirSync(stateDirectory)
+		const coordinatorFiles = files.filter(
+			(file) => file.startsWith('coordinator-db-') && file.endsWith('.json')
+		)
+
+		return coordinatorFiles.map((file) => {
+			// Extract program ID from filename: coordinator-db-{programId}.json
+			const programId = file.slice('coordinator-db-'.length, -'.json'.length)
+			return programId
```

Comment:
it could be sligthly easier to follow if you make the mapping first storing it in a variable and then returning it